src/MerQure.Tools/Buses/Publisher.cs

```diff
@@ -0,0 +1,134 @@
+            }
+        }
+
+        public void Publish(Channel channel, T message, bool applyDeliveryDeplay)
+        {
+            var encapsuledMessage = CreateRetryMessage(message);
+            string bindingValue = channel.Value;
+            string busName = _messageBrokerConfiguration.BusName;
+            if (applyDeliveryDeplay && _messageBrokerConfiguration.DeliveryDelayInMilliseconds > 0)
+            {
+                bindingValue = $"{bindingValue}.{_messageBrokerConfiguration.DeliveryDelayInMilliseconds}";
+                busName = $"{busName}.{ RetryStrategyConfiguration.RetryExchangeSuffix}";
+            }
+            using (var publisher = _messagingService.GetPublisher(busName, true))
+            {
+                TryPublishWithBrokerAcknowledgement(publisher, bindingValue, JsonConvert.SerializeObject(encapsuledMessage));
+            }
+        }
+        
+        public void PublishOnRetryExchange(Channel channel, T message, RetryInformations retryInformations)
+        {
+            List<int> delays = _messageBrokerConfiguration.DelaysInMsBetweenEachRetry;
+            int delay = 0;
+            if (delays.Count() >= retryInformations.NumberOfRetry)
+            {
+                delay = delays[retryInformations.NumberOfRetry];
+            }
+            else
+            {
+                delay = delays.Last();
+            }
+            retryInformations.NumberOfRetry++;
+            RetryMessage<T> retryMessage = new RetryMessage<T>
+            {
+                OriginalMessage = message,
+                RetryInformations = retryInformations
+            };
+
+            string bindingValue = $"{channel.Value}.{delay}";
+            using (var publisher = _messagingService.GetPublisher($"{_messageBrokerConfiguration.BusName}.{RetryStrategyConfiguration.RetryExchangeSuffix}", true))
+            {
+                TryPublishWithBrokerAcknowledgement(publisher, bindingValue, JsonConvert.SerializeObject(retryMessage));
+            }
+        }
+
+        internal void PublishOnErrorExchange(Channel channel, T message, RetryInformations technicalInformations)
+        {
+            string errorChanel = $"{channel.Value}.error";
+            RetryMessage<T> retryMessage = new RetryMessage<T>
+            {
+                OriginalMessage = message,
+                RetryInformations = technicalInformations
+            };
+            using (var publisher = _messagingService.GetPublisher($"{_messageBrokerConfiguration.BusName}.{RetryStrategyConfiguration.ErrorExchangeSuffix}", true))
+            {
+                TryPublishWithBrokerAcknowledgement(publisher, errorChanel, JsonConvert.SerializeObject(retryMessage));
+            }
+        }
+
+        internal void TryPublishWithBrokerAcknowledgement(IPublisher publisher, string channelName, string message)
+        {
+            bool published = publisher.PublishWithAcknowledgement(channelName, message);
+            if (published == false)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the unnecessary Boolean literal(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS1125)